rfc/0020-group-policies.md

```diff
@@ -0,0 +1,263 @@
+|              |                                 |
+| :----------- | :------------------------------ |
+| Feature Name | Group policies                  |
+| Start Date   | Jun 12 2024                     |
+| Category     | feature                         |
+| RFC PR       | [fill this in after opening PR] |
+| State        | **ACCEPTED**                    |
+
+# Summary
+
+A "group policy" is a policy that is composed of other policies.
+The policies that are part of a group are evaluated using a boolean expression.
+The group policy will not support mutations.
+
+# Motivation
+
+[motivation]: #motivation
+
+The motivation for this feature is to enable users to create complex policies by combining simpler ones. This allows users to avoid the need to create custom policies from scratch and instead leverage existing policies.
+This reduces the need to duplicate policy logic across different policies, increases reusability, removes the cognitive load of managing complex policy logic,
+and enables the creation of custom policies using a DSL-like configuration.
+
+## Examples / User Stories
+
+[examples]: #examples
+
+- As a user, I want to create a policy that is composed of other policies and is evaluated using a boolean expression.
+
+# Detailed design
+
+[design]: #detailed-design
+
+## Controller and CRDs
+
+This RFC proposes to add two new CRDs to the Kubewarden controller:
+
+- `GroupAdmissionPolicy`
+- `GroupClusterAdmissionPolicy`
+
+The CRDs will share most of the fields with the `AdmissionPolicy` and `ClusterAdmissionPolicy` CRDs, respectively.
+The main difference is that the `GroupAdmissionPolicy` and `GroupClusterAdmissionPolicy` CRDs will have a list of policies.
+Also, the `GroupAdmissionPolicy` and `GroupClusterAdmissionPolicy` CRDs will have a field to specify the boolean expression that will be used to evaluate the policies
+and a `message` field to specify the message that will be returned when the group policy is rejected.
+
+## Example CRD
+
+```yaml
+apiVersion: policies.kubewarden.io/v1
+kind: ClusterAdmissionPolicyGroup # or AdmissionPolicyGroup
```

Comment:
This naming is inconsistent with the previous paragraph. I highly prefer this one though.
In any case, I would prefer one that starts with `Admission` and `ClusterAdmission` rather than `Group` and `GroupCluster`.

Reply:
I also prefer `ClusterAdmissionPolicyGroup`

Reply:
> I also prefer ClusterAdmissionPolicyGroup

Same here